prime-router/metadata/fhir_mapping/hl7/datatypes/CX/Identifier.yml

```diff
@@ -0,0 +1,177 @@
+      value: identifier-check-digit
+    valueString:
+      type: STRING
+      expressionType: HL7Spec
+      valueOf: CX.2
+
+extension_2:
+  generateList: true
+  expressionType: nested
+  expressionsMap:
+    url:
+      type: SYSTEM_URL
+      value: naming-system-check-digit
+    valueString:
+      type: STRING
+      expressionType: HL7Spec
+      valueOf: CX.3
+
+# Checking if HD is an Organization or a System by looking at the ID type.
+# If ID type is null or ID type is not ISO or UUID, treat as organization.
+# Otherwise, treat as system.
+assigner_1:
+  vars:
+    cx43: String, CX.4.3
+  condition: cx43 NULL
+  valueOf: datatypes/HD/Organization
+  expressionType: reference
+  specs: CX.4
+
+assigner_2:
+  vars:
+    cx43: String, CX.4.3
+  condition: $cx43 NOT_NULL && $cx43 NOT_EQUALS ISO && $cx43 NOT_EQUALS UUID
+  valueOf: datatypes/HD/Organization
+  expressionType: reference
+  specs: CX.4
+
+system:
+  vars:
+    cx41: String, CX.4.1
+    cx43: String, CX.4.3
+  expressionType: nested
+  condition: $cx43 EQUALS ISO || cx43 EQUALS UUID
+  expressions:
+    - condition: $cx41 NOT_NULL
+      expressionType: HL7Spec
+      type: STRING
+      valueOf: CX.4.1
+    - condition: $cx41 NULL && cx43 EQUALS ISO
+      expressionType: HL7Spec
+      type: STRING
+      constants:
+        prefix: "urn:oid:"
+      vars:
+        uri: String, CX.4.2
+        join: $prefix + $uri
+      valueOf: $join
+    - condition: $cx41 NULL && cx43 EQUALS UUID
+      expressionType: HL7Spec
+      type: STRING
+      constants:
+        prefix: "urn:uuid:"
+      vars:
+        uri: String, CX.4.2
+        join: $prefix + $uri
+      valueOf: $join
+
+extension_system:
+  generateList: true
+  expressionType: nested
+  expressionsMap:
+    url:
+      type: SYSTEM_URL
+      value: coding-system
+    valueCodeableConcept:
+      valueOf: datatype/CodeableConcept_var
+      expressionType: resource
+      vars:
+        code: String, CX.4
+
+type:
+  valueOf: datatype/CodeableConcept_var
+  expressionType: resource
+  vars:
+    code: $code
+    system: $system
+    text: $text
+
+extension_assigningFacility:
+  generateList: true
+  expressionType: nested
+  expressionsMap:
+    url:
+      type: SYSTEM_URL
+      value: assigning-facility
+    valueReference:
+      valueOf: datatypes/HD/Organization
+      expressionType: reference
+      specs: CX.6
+
+period:
+  valueOf: datatype/Period
+  expressionType: resource
+  vars:
+    start: CX.7
+    end: CX.8
+
+extension_4:
+  generateList: true
+  expressionType: nested
+  expressionsMap:
+    url:
+      type: SYSTEM_URL
+      value: assigning-jurisdiction
+    valueString:
+      type: STRING
+      expressionType: HL7Spec
+      valueOf: CX.9
+
+extension_5:
+  generateList: true
+  expressionType: nested
+  expressionsMap:
+    url:
+      type: SYSTEM_URL
+      value: assigning-agency-or-department
+    valueString:
+      type: STRING
+      expressionType: HL7Spec
+      valueOf: CX.10
```

Comment:
CX.9 and CX.10 are CWEs, so I don't think we can handle them as strings, we probably need to put a codeable concept on these extensions for it to work.

Reply:
Might be good to enhance the tests to have more than just CX.9.1/CX.10.1 filled too.